Add Ecmascript stage3 proposal support to acorn

Please add https://github.com/acornjs/acorn-stage3 support to acorn so that it doesn't fail on modern javascript. It could of course be configurable.

Hi and sorry for my previous short message.

I use the [dynamic-import proposal](https://github.com/tc39/proposal-dynamic-import) feature like this in the `<script>` portion of my Vue component:
```js
const Detector = () => import("./Detector.vue");
```
The parsing issue happens just before the `import`.
```
Trace: { SyntaxError: Unexpected token (58:23)
    at Parser.pp$4.raise ([...]/node_modules/acorn/dist/acorn.js:2756:13)
    at Parser.pp.unexpected ([...]/node_modules/acorn/dist/acorn.js:647:8)
    at Parser.pp$3.parseExprAtom ([...]/node_modules/acorn/dist/acorn.js:2196:10)
    at Parser.pp$3.parseExprSubscripts ([...]/node_modules/acorn/dist/acorn.js:2047:19)
    at Parser.pp$3.parseMaybeUnary [...]/node_modules/acorn/dist/acorn.js:2024:17)
    at Parser.pp$3.parseExprOps ([...]/node_modules/acorn/dist/acorn.js:1966:19)
    at Parser.pp$3.parseMaybeConditional ([...]/node_modules/acorn/dist/acorn.js:1949:19)
    at Parser.pp$3.parseMaybeAssign ([...]/node_modules/acorn/dist/acorn.js:1925:19)
    at Parser.pp$3.parseFunctionBody ([...]/node_modules/acorn/dist/acorn.js:2577:22)
    at Parser.pp$3.parseArrowExpression ([...]/node_modules/acorn/dist/acorn.js:2560:8)
    at Parser.pp$3.parseParenArrowList ([...]/node_modules/acorn/dist/acorn.js:2283:15)
    at Parser.pp$3.parseParenAndDistinguishExpression ([...]/node_modules/acorn/dist/acorn.js:2249:19)
    at Parser.pp$3.parseExprAtom ([...]/node_modules/acorn/dist/acorn.js:2163:41)
    at Parser.pp$3.parseExprSubscripts ([...]/node_modules/acorn/dist/acorn.js:2047:19)
    at Parser.pp$3.parseMaybeUnary ([...]/node_modules/acorn/dist/acorn.js:2024:17)
    at Parser.pp$3.parseExprOps ([...]/node_modules/acorn/dist/acorn.js:1966:19)
  pos: 290,
  loc: Position { line: 58, column: 23 },
  raisedAt: 296 }
    at /[...]/node_modules/easygettext/src/extract-cli.js:71:13
    at Array.forEach (<anonymous>)
    at Object.<anonymous> (/[...]/node_modules/easygettext/src/extract-cli.js:57:7)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
Makefile:34 : la recette pour la cible « /tmp/template.pot » a échouée
make: *** [/tmp/template.pot] Erreur 1
```

--- a/package.json
+++ b/package.json
@@ -32,6 +32,7 @@
   "dependencies": {
     "@vue/component-compiler-utils": "^1.2.1",
     "acorn": "^5.5.3",
+    "acorn-stage3": "^0.6.0",
     "cheerio": "^1.0.0-rc.2",
     "minimist": "^1.2.0",
     "pofile": "^1.0.10",

--- a/src/javascript-extract.js
+++ b/src/javascript-extract.js
@@ -1,4 +1,4 @@
-const acorn = require('acorn');
+const acorn = require('acorn-stage3');
 const Pofile = require('pofile');
 
 const {MARKER_NO_CONTEXT, DEFAULT_VUE_GETTEXT_FUNCTIONS} = require('./constants.js');
@@ -29,6 +29,9 @@ function getGettextEntriesFromScript(script) {
     sourceType: 'module',
     locations: true,
     onToken: allTokens,
+    plugins: {
+      stage3: true
+    },
   };
 
   acorn.parse(script, ACORN_OPTIONS);

--- a/src/javascript-extract.spec.js
+++ b/src/javascript-extract.spec.js
@@ -101,5 +101,16 @@ describe('Javascript extractor object', () => {
       expect(extractedStrings.length).to.be.equal(1);
       expect(extractedStrings[0].msgid).to.be.equal('Hello world from the $gettext function');
     });
+
+    it('should not break parser when using ECMAScript Stage 3 features', () => {
+      const filename = 'stage3.vue';
+      const extractedStrings = jsExtractor.extractStringsFromJavascript(
+        filename,
+        fixtures.SCRIPT_WITH_ES_STAGE3_FEATURES
+      );
+
+      expect(extractedStrings.length).to.be.equal(1);
+      expect(extractedStrings[0].msgid).to.be.equal('Hello world from the future');
+    });
   });
 });

--- a/src/test-fixtures.js
+++ b/src/test-fixtures.js
@@ -327,6 +327,11 @@ export default {
   }
 }`;
 
+exports.SCRIPT_WITH_ES_STAGE3_FEATURES = `
+const asyncModule = () => import('module');
+const message = this.$gettext('Hello world from the future');
+`;
+
 exports.SCRIPT_GETTEXT_SEQUENCE_FILENAME = 'gettext_sequence.vue';
 exports.SCRIPT_GETTEXT_SEQUENCE = `
 export default {

